presto-spi/src/main/java/com/facebook/presto/spi/PrestoSparkExecutionContext.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrestoSparkExecutionContext
+{
+    private final PhysicalResourceSettings physicalResourceSettings;
+
+    @JsonCreator
+    public PrestoSparkExecutionContext(@JsonProperty("physicalResourceSettings") PhysicalResourceSettings physicalResourceSettings)
+    {
+        this.physicalResourceSettings = requireNonNull(physicalResourceSettings, "physicalResourceSettings is null");
+    }
+
+    @JsonProperty
+    public PhysicalResourceSettings getPhysicalResourceSettings()
+    {
+        return physicalResourceSettings;
+    }
+
+    public static PrestoSparkExecutionContext create(
```

Comment:
I suppose we will have more such static function to add new attributes to `PrestoSparkExecutionContext`.
In that case, should this function be called `.withResourceSettings`, like a builder pattern.

Reply:
@vermapratyush : I was thinking more in terms of just extending the method signature with new attributes and then create and return an instance of PrestoSparkExecutionContext. Something similar to `QueryStats.create()`